### PR TITLE
Module 2

### PR DESCRIPTION
**Pull request by kelechi-chima**
Module 2 exercise.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -32,6 +32,9 @@ $ cp .env.template .env  # (first time only)
 
 The `.env` file is used by flask to set environment variables when running `flask run`. This enables things like development mode (which also enables features like hot reloading when you make a file change). There's also a [SECRET_KEY](https://flask.palletsprojects.com/en/1.1.x/config/#SECRET_KEY) variable which is used to encrypt the flask session cookie.
 
+## Environment variables
+The app reads TRELLO_API_KEY and TRELLO_API_TOKEN from the .env file. You will need to register on Trello to get your API key and token.
```
Comment (BenRamchandani) [inline README.md:36]: Can you add placeholders for these to your `.env.template` file?

File: pyproject.toml [hunk 2]
```diff
@@ -8,6 +8,7 @@ authors = []
 python = "^3.7"
 flask = "^1.1.2"
 python-dotenv = "^0.14.0"
+requests = "^2.25.1"
```
Comment (BenRamchandani) [inline pyproject.toml:11]: It looks like you haven't committed the changes to your `poetry.lock` file. 

Try doing:
```bash
rm -r .venv/
poetry install
```
I expect you'll get an error like I did running `poetry install`. Try running `poetry update` and then committing the changes to `poetry.lock`

File: todo_app/app.py [hunk 3]
```diff
@@ -3,31 +3,28 @@
...
 
 
-@app.route('/complete', methods=['POST'])
-def complete_item():
-    completed = request.form.getlist('completed')
-    for c in completed:
-        item = get_item(c)
-        item['status'] = 'Completed'
-        save_item(item)
+@app.route('/complete/<item_id>', methods=['GET'])
```
Comment (BenRamchandani) [inline todo_app/app.py]: Why have you changed this to a GET route? It changes the state of your app so it is better to use POST

File: todo_app/trello_api.py [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+    url = 'https://api.trello.com/1/boards/'
+    query = dict(base_query)
+    query['name'] = board_name
+    response = requests.post(url, params=query)
+    result = response.json()
+    return result['id']
+
+def get_boards():
+    username = os.environ.get('TRELLO_USERNAME')
+    url = f'https://api.trello.com/1/members/{username}/boards'
```
Comment (BenRamchandani) [inline todo_app/trello_api.py]: You can use `https://api.trello.com/1/members/me/boards`, that way you don't need to know the username

**Review by BenRamchandani — CHANGES_REQUESTED** [review]
Hey, this looks good so far, but you've missed Part 5 of the exercise "Create a class for to-do items". Can you have a go at that?

I've left a few other comments as well which should hopefully be easy changes.

File: todo_app/app.py (hunk 3 above)
Comment (kelechi-chima) [reply todo_app/app.py]: Sorry I resolved this on the 27th of May but didn't click the Resolve conversation button.

**Review by BenRamchandani — APPROVED** [review]
Looks great @kelechi-chima, thanks for doing the markups.

Let me know once you've resubmitted it on Aptem and I'll approve it on there too.

**Comment by kelechi-chima** [comment]
> Looks great @kelechi-chima, thanks for doing the markups.
> 
> Let me know once you've resubmitted it on Aptem and I'll approve it on there too.

Hi @BenRamchandani, I've resubmitted it on Aptem. Please take a look.

**Comment by BenRamchandani** [comment]
All done, merge it in when you're ready 🙂

**Pull request merged**